applications/vanilla/controllers/class.postcontroller.php

```diff
@@ -351,7 +351,7 @@ public function discussion($CategoryUrlCode = '') {
                         $this->fireEvent('AfterDiscussionSave');
 
                         if ($this->_DeliveryType == DELIVERY_TYPE_ALL) {
-                            redirect(discussionUrl($Discussion, 1)).'?new=1';
+                            redirectTo(discussionUrl($Discussion, 1), 302, false).'?new=1';
```

Comment:
This isn't behavior you introduced, but it appears incorrect. The `new` param is being appended to the result of `redirectTo`.  It should be appended to the result of `discussionUrl`.

```php
redirectTo(discussionUrl($Discussion, 1).'?new=1', 302, false);
```